crates/services/txpool/src/service.rs

```diff
@@ -136,12 +138,29 @@ where
                 should_continue = false;
             }
             new_transaction = self.gossiped_tx_stream.next() => {
-                if let Some(GossipData { data: Some(tx), .. }) = new_transaction {
+                if let Some(GossipData { data: Some(_), .. }) = &new_transaction {
+                    // safety: both Options have been checked with above `if let` statement
+                    let mut new_transaction = new_transaction.unwrap();
+                    let tx = new_transaction.take_data().unwrap();
                     let txs = vec!(Arc::new(tx));
-                    self.shared.txpool.lock().insert(
+
+                    let mut result = self.shared.txpool.lock().insert(
                         &self.shared.tx_status_sender,
                         &txs
                     );
+
+                    if let Some(acceptance) = match result.pop() {
```

Comment:
I would prefer to add `expect` here because we expect that `result.len() == 1`, in this case you can use `if else` statement instead of `match`

Reply:
alternately, maybe just adding a helper fn to convert any kind of `Ok` or `Err` into an `Accept` or `Reject` would be enough to clean up this code without risking any runtime panic.